osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyImportedScoresCommand.cs

```diff
@@ -0,0 +1,201 @@
+                {
+                    var rulesetSpecifics = LegacyDatabaseHelper.GetRulesetSpecifics(rulesetScores.Key);
+
+                    var highScores = await conn.QueryAsync<HighScore>($"SELECT * FROM {rulesetSpecifics.HighScoreTable} WHERE score_id IN ({string.Join(',', rulesetScores.Select(s => s.legacy_score_id))})");
+
+                    var legacyMaps = await conn.QueryAsync<SoloScoreLegacyIDMap>($"SELECT * FROM score_legacy_id_map WHERE ruleset_id = {rulesetScores.Key} AND old_score_id IN ({string.Join(',', rulesetScores.Select(s => s.legacy_score_id))})");
+
+                    foreach (var score in rulesetScores)
+                        score.HighScore = highScores.SingleOrDefault(s => s.score_id == score.legacy_score_id!.Value);
+
+                    foreach (var score in rulesetScores)
+                        score.LegacyMap = legacyMaps.SingleOrDefault(s => s.old_score_id == score.legacy_score_id!.Value);
+                }
+
+                if (!importedScores.Any())
+                {
+                    Console.WriteLine("All done!");
+                    break;
+                }
+
+                elasticItems.Clear();
+
+                foreach (var importedScore in importedScores)
+                {
+                    if (importedScore.legacy_score_id == null) continue;
+
+                    if (importedScore.HighScore == null)
+                    {
+                        await conn.ExecuteAsync("DELETE FROM scores WHERE id = @newId; DELETE FROM score_performance WHERE score_id = @newId; DELETE FROM score_legacy_id_map WHERE old_score_id = @oldId AND ruleset_id = @rulesetId", new
+                        {
+                            newId = importedScore.id,
+                            oldId = importedScore.legacy_score_id,
+                            rulesetId = importedScore.ruleset_id,
+                        });
+
+                        Interlocked.Increment(ref deleted);
+                        continue;
+                    }
+
+                    if (importedScore.LegacyMap == null || importedScore.LegacyMap.score_id != importedScore.id)
+                    {
+                        Console.WriteLine($"{importedScore.id}: Legacy mapping entry missing or incorrect!!");
+                        Interlocked.Increment(ref fail);
+
+                        await conn.ExecuteAsync($"REPLACE INTO score_legacy_id_map (ruleset_id, old_score_id, score_id) VALUES ({importedScore.ruleset_id}, {importedScore.legacy_score_id}, {importedScore.id})");
+                    }
+
+                    if (importedScore.pp == null && importedScore.HighScore.pp != null)
+                    {
+                        Console.WriteLine($"{importedScore.id}: Performance entry missing!!");
+                        Interlocked.Increment(ref fail);
+
+                        await conn.ExecuteAsync($"REPLACE INTO score_performance VALUES ({importedScore.id}, {importedScore.HighScore.pp})");
```

Comment:
The lack of an explicit column name list is a *bit* scary here but from testing on my local env it does seem like this should work as expected in this order...